File: 1869. Longer Contiguous Segments of Ones than Zeros/solution.py
class Solution(object):
  def checkZeroOnes(self, s):
    """
    :type s: str
    :rtype: bool
    """
    currentZeros = 0
    maxZeros = 0
    currentOnes = 0
    maxOnes = 0
    
    for number in s:
      if number == '0':
        currentZeros += 1
        currentOnes = 0
      elif number == '1':
        currentOnes += 1
        currentZeros = 0
        
      if maxZeros < currentZeros:
        maxZeros = currentZeros
        
      if maxOnes < currentOnes:
        maxOnes = currentOnes
          
    return maxOnes > maxZeros

File: 1869. Longer Contiguous Segments of Ones than Zeros/solution.cpp
class Solution {
public:
  bool checkZeroOnes(string s) {
    int currentZeros = 0;
    int maxZeros = 0;
    int currentOnes = 0;
    int maxOnes = 0;

    for (int index = 0; index < s.size(); index++){
      char number = s[index];
      if (number == '0'){
        currentZeros += 1;
        currentOnes = 0;
      }
      else if (number == '1'){
        currentOnes += 1;
        currentZeros = 0;
      }

      if (maxZeros < currentZeros){
        maxZeros = currentZeros;
      }

      if (maxOnes < currentOnes){
        maxOnes = currentOnes;
      }
    }

    return maxOnes > maxZeros;
  }
};